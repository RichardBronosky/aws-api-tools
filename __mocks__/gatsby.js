const React = require("react")
const gatsby = jest.requireActual("gatsby")

// mock matchMedia for react-slick
window.matchMedia = jest.fn().mockImplementation(query => {
  return {
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
  }
})

// gatsby testing setup from graphql, link, StaticQuery
module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      to,
      ...rest
    }) =>
      React.createElement("a", {
        ...rest,
        href: to,
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}
