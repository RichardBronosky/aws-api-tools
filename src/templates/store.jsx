import React from 'react'
import Helmet from 'react-helmet'
import { Map, Marker, InfoWindow, GoogleApiWrapper } from 'google-maps-react'
import Layout from '../components/layout'
import Breadcrumb from '../components/shared/breadcrumbs'
import RTGLink from '../components/shared/link'
import { decodeHtml, slugify, weekdays, months, getStandardTime } from '../lib/helpers/string-helper'
import { contentfulImage } from '../lib/helpers/contentful'
import { getStoreHours, getStoreName, getStoreInfoFromFeed } from '../lib/helpers/store-locator'
import { expandState } from '../lib/helpers/geo-location'
import { setupAnalytics } from '../lib/helpers/google-tag-manager'
import StoreInfo from '../components/store-locator/store-info'
import '../assets/css/pages/store-location.sass'

class Store extends React.Component {
  state = {
    showingInfoWindow: false,
    activeMarker: {},
    selectedMarker: {},
  }

  componentDidMount() {
    const store = this.props.pageContext.store
    setupAnalytics({ pageData: { type: 'store', title: getStoreName(store), path: window.location.pathname } })
  }

  createCrumb = (crumb, slugPath) => {
    return {
      altDesc: `View All ${ crumb } Products`,
      id: slugPath,
      slug: slugPath,
      text: crumb,
    }
  }

  generateStoreCrumbs = store => {
    let crumbs = []
    crumbs.push(this.createCrumb('Find a Showroom', '/stores'))
    if (store.state) {
      crumbs.push(
        this.createCrumb(
          `Find a Showroom - ${ expandState(store.state) }`,
          `/stores/${ slugify(expandState(store.state)) }`
        )
      )
    }
    crumbs.push(
      this.createCrumb(
        decodeHtml(`${ store.city }, ${ store.state }${ store.storeName ? ` (${ store.storeName })` : '' }`),
        window.location.pathname
      )
    )
    return {
      crumbs: crumbs,
    }
  }

  onLocationClick = (props, marker) => {
    const store = this.props.pageContext.store
    const storeInfoFromFeed = getStoreInfoFromFeed(store.storeNumber)
    let fullMarker = {
      ...props,
      title: store.address1,
      name: getStoreName(store),
      lat: store.location.lat,
      lng: store.location.lon,
      city: store.city,
      state: store.state,
      storeNumber: store.storeNumber,
      zip: store.zip,
      phone: storeInfoFromFeed.phone.replace('/', '-'),
      hours: getStoreHours(storeInfoFromFeed.store_hours),
    }
    this.setState({
      selectedMarker: fullMarker,
      activeMarker: marker,
      showingInfoWindow: true,
    })
  }

  onMapClicked = () => {
    if (this.state.showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: null,
      })
    }
  }

  getTodayOpen = store => {
    const currentDay = weekdays[new Date().getDay()]
    const month = months[new Date().getMonth()].substr(0, 3)
    const day = new Date().getDate()
    const storeOpen = store.store_hours[`${ currentDay.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ currentDay.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ currentDay }, ${ month } ${ day }: ${
        storeOpen ? `Open from ${ getStandardTime(storeOpen) } to ${ getStandardTime(storeClose) }` : 'Closed'
      }`
    } else {
      ;`${ currentDay }, ${ month } ${ day }`
    }
  }

  getHours = (store, day) => {
    const storeOpen = store.store_hours[`${ day.toLowerCase() }Open`]
    const storeClose = store.store_hours[`${ day.toLowerCase() }Closed`]
    if (storeOpen && storeClose) {
      return `${ getStandardTime(storeOpen) } - ${ getStandardTime(storeClose) }`
    }
  }

  render() {
    const { activeMarker, selectedMarker, showingInfoWindow } = this.state
    const store = this.props.pageContext.store
    const address = `${ store.address1 }${ store.address2 ? `, ${ store.address2 }` : '' }, ${ store.city }, ${ store.state } ${
      store.zip
    } `
    const storeInfoFromFeed = getStoreInfoFromFeed(store && store.storeNumber)
    if (store) {
      let storeTypeTitle = `${ store.city }, ${ store.state }`
      if (store.storeType === 'Showroom' || store.storeType === 'Express') {
        storeTypeTitle = storeTypeTitle + ' Furniture & Mattress Store'
      } else if (store.storeType === 'Kids') {
        storeTypeTitle = storeTypeTitle + ' Kids Furniture & Mattress Store'
      } else if (store.storeType === 'Outlet') {
        storeTypeTitle = storeTypeTitle + ' Affordable Furniture Outlet Store'
      } else if (store.storeType === 'Outdoor') {
        storeTypeTitle = storeTypeTitle + ' Patio Furniture: Wicker, Teak Outdoor Furniture Sets'
      }
      return (
        <Layout { ...this.props }>
          <Helmet title={ storeTypeTitle } />
          { store.seo && (
            <>
              { store.seo.pageTitle && <Helmet title={ store.seo.pageTitle } /> }
              { store.seo.canonical && (
                <Helmet
                  link={ [
                    {
                      rel: 'canonical',
                      href: store.seo.canonical,
                    },
                  ] }
                />
              ) }
              { store.seo.metaDescription && (
                <Helmet
                  meta={ [
                    {
                      name: 'description',
                      content: store.seo.metaDescription.metaDescription,
                    },
                  ] }
                />
              ) }
              { store.seo.metaRobots && (
                <Helmet
                  meta={ [
                    {
                      name: 'robots',
                      content: store.seo.metaRobots,
                    },
                  ] }
                />
              ) }
              { store.seo.jsonLdSchema && (
                <Helmet
                  script={ [
                    {
                      type: 'application/ld+json',
                      innerHTML: store.seo.jsonLdSchema.jsonLdSchema,
                    },
                  ] }
                />
              ) }
            </>
          ) }
          <section className="seo-page-heading full-width hide show">
            <div className="grid-container ">
              <div className="grid-x grid-margin-x">
                <div className="cell small-12">
                  <h1>
                    { store.seo && store.seo.pageHeading
                      ? store.seo.pageHeading
                      : `${ storeTypeTitle }${ store.storeName ? ` (${ store.storeName })` : '' }` }
                  </h1>
                </div>
              </div>
            </div>
          </section>
          <div className="stores-breadcrumb">
            <Breadcrumb data={ this.generateStoreCrumbs(store) } includeHeading={ true } />
          </div>
          <div className="store-location">
            <div className="loc-info">
              { address }
              <RTGLink
                data={ {
                  url: `tel:${ storeInfoFromFeed.phone.replace('/', '-') }`,
                  title: 'Call store',
                  category: 'store page',
                  action: 'Call click',
                  label: 'Call',
                } }
                className="phone-link"
              >
                { storeInfoFromFeed.phone.replace('/', '-') }
              </RTGLink>
              <span className="today">{ this.getTodayOpen(storeInfoFromFeed) }</span>
            </div>
            <div className="grid-x map-container">
              <Map
                initialCenter={ { lat: store.location.lat, lng: store.location.lon } }
                center={ { lat: store.location.lat, lng: store.location.lon } }
                className="map"
                google={ this.props.google }
                zoom={ 11 }
                onClick={ this.onMapClicked }
              >
                <Marker
                  position={ { lat: store.location.lat, lng: store.location.lon } }
                  title={ `${ store.city }${ store.storeName ? ` - ${ store.storeName }` : '' } ${ store.storeType }` }
                  icon={ {
                    url: `http://maps.google.com/mapfiles/markerA.png`,
                  } }
                  onClick={ this.onLocationClick }
                />
                <InfoWindow marker={ activeMarker } visible={ showingInfoWindow }>
                  <StoreInfo selectedMarker={ selectedMarker } />
                </InfoWindow>
              </Map>
              { store.storeImage && (
                <div className="store-image-container">
                  <img className="store-image" alt={ storeTypeTitle } src={ contentfulImage(store.storeImage.file.url) } />
                </div>
              ) }
              { storeInfoFromFeed.store_hours && (
                <div className="section-container">
                  <h2 className="section-heading">Store Hours:</h2>
                  { weekdays.map(day => (
                    <div className="hour" key={ day }>
                      { storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">{ this.getHours(storeInfoFromFeed, day) }</span>
                        </>
                      ) }
                      { !storeInfoFromFeed.store_hours[`${ day.toLowerCase() }Open`] && (
                        <>
                          <span className="day">{ day }</span>
                          <span className="time">Closed</span>
                        </>
                      ) }
                    </div>
                  )) }
                  { store.specialClosings && (
                    <span
                      className="special-closings"
                      dangerouslySetInnerHTML={ { __html: store.specialClosings.markdown.childMarkdownRemark.html } }
                    />
                  ) }
                </div>
              ) }
              { store.directions && (
                <div className="section-container">
                  <h2 className="section-heading">Store Location / Address:</h2>
                  <span>{ address }</span>
                  <br />
                  <span
                    className="directions"
                    dangerouslySetInnerHTML={ { __html: store.directions.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.neighborhoodsNearby && store.neighborhoodsNearby.length > 0 && (
                <div className="section-container">
                  <h2 className="section-heading">Neighborhoods Nearby:</h2>
                  <ul>
                    { store.neighborhoodsNearby.map((hood, index) => (
                      <li key={ index } className="hood">
                        { hood }
                      </li>
                    )) }
                  </ul>
                </div>
              ) }
              { store.description && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">Store Description:</h2>
                  <span
                    className="description"
                    dangerouslySetInnerHTML={ { __html: store.description.childMarkdownRemark.html } }
                  />
                </div>
              ) }
              { store.faqs && (
                <div className="section-container fullwidth">
                  <h2 className="section-heading">FAQs:</h2>
                  <span
                    className="faqs"
                    dangerouslySetInnerHTML={ { __html: store.faqs.markdown.childMarkdownRemark.html } }
                  />
                </div>
              ) }
            </div>
          </div>
        </Layout>
      )
    } else {
      return null
    }
  }
}

export default GoogleApiWrapper({
  apiKey: `${ process.env.GATSBY_GOOGLE_BROWSER_API_KEY }`,
})(Store)
