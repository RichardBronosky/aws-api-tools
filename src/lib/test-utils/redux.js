import configureMockStore from 'redux-mock-store'
import * as reduxStore from '../../redux/store'

const mockStore = configureMockStore([])

export const mockStoreDispatch = initialState => {
  reduxStore.store = mockStore(initialState)
  reduxStore.store.dispatch = jest.fn()
}
