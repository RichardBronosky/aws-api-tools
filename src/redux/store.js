import React from 'react'
import { Provider } from 'react-redux'
import { createStore, compose, combineReducers } from 'redux'

import global from './modules/global'
import location from './modules/location'
import cart from './modules/cart'
import checkout from './modules/checkout'

const rootReducer = combineReducers({
  global,
  location,
  cart,
  checkout,
})

// Redux DevTools
const windowObject = typeof window !== 'undefined' && window
const reduxDevTools =
  process.env.NODE_ENV === 'development' && windowObject.devToolsExtension
    ? window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    : _ => _

const configureStore = initialState => createStore(rootReducer, initialState, compose(reduxDevTools))

export const store = configureStore()

// eslint-disable-next-line react/display-name,react/prop-types
export default ({ element }) => <Provider store={ store }>{ element }</Provider>
