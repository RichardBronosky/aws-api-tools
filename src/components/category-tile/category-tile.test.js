import React from 'react'
import renderer from 'react-test-renderer'
import { StaticQuery } from 'gatsby'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'
import CategoryTile from './category-tile'

const categoryTileData = {
  name: 'test-category-tile',
  image: {
    file: {
      url: '/test-image',
    },
  },
  mobileImage: {
    file: {
      url: '/test-image-mobile',
    },
  },
  link: {
    text: 'test category tile link',
  },
  collapsed: false,
  collapsedImageTop: false,
  onSale: false,
}

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      contentfulCategoryTile: categoryTileData,
    })
  )
})

describe('CategoryTile', () => {
  const testProps = {
    isMobile: false,
    category: categoryTileData,
  }

  it('renders correctly with default props', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...testProps } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is true', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ ...testProps, isMobile: true } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is true and there is no mobile image', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ category: { ...categoryTileData, mobileImage: null }, isMobile: true } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is false and there is no mobile image', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ category: { ...categoryTileData, mobileImage: null } } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is true and there is no image or mobile image', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ category: { ...categoryTileData, mobileImage: null, image: null }, isMobile: true } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is false and is collapsed', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ category: { ...categoryTileData, collapsed: true } } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is false and is collapsed image top', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ category: { ...categoryTileData, collapsedImageTop: true } } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is false and is collapsed and collapsed image top', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ category: { ...categoryTileData, collapsed: true, collapsedImageTop: true } } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when isMobile is false and is on sale', () => {
    const tree = renderer
      .create(
        <Provider store={ store }>
          <CategoryTile { ...{ category: { ...categoryTileData, onSale: true } } } />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
