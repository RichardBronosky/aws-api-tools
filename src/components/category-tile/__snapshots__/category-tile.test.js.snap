// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTile renders correctly when isMobile is false and is collapsed 1`] = `
<div
  className="category-tile category-small"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image?cache-id=&h=250"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly when isMobile is false and is collapsed and collapsed image top 1`] = `
<div
  className="category-tile category-small category-small-image-top"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image?cache-id=&h=250"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly when isMobile is false and is collapsed image top 1`] = `
<div
  className="category-tile category-small-image-top"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image?cache-id=&h=550"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly when isMobile is false and is on sale 1`] = `
<div
  className="category-tile"
>
  <span
    className=" sale-flag"
    onClick={[Function]}
  >
    Sale
  </span>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image?cache-id=&h=550"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly when isMobile is false and there is no mobile image 1`] = `
<div
  className="category-tile"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image?cache-id=&h=550"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly when isMobile is true 1`] = `
<div
  className="category-tile"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image-mobile?cache-id=&h=350"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly when isMobile is true and there is no image or mobile image 1`] = `
<div
  className="category-tile"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src=""
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly when isMobile is true and there is no mobile image 1`] = `
<div
  className="category-tile"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image?cache-id=&h=550"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;

exports[`CategoryTile renders correctly with default props 1`] = `
<div
  className="category-tile"
>
  <span
    className="category-image"
  >
    <img
      alt=""
      aria-hidden="true"
      role="presentation"
      src="undefinedtest-image?cache-id=&h=550"
    />
  </span>
  <div
    className="category-name"
    style={
      Object {
        "color": "#333",
      }
    }
  >
    test-category-tile
  </div>
</div>
`;
